ironfist/src/cpp/game/save.cpp

```diff
@@ -144,23 +144,27 @@ void ReadHeroXML(ironfist_map::hero_t& hx, hero* hro) {
 	hro->y = hx.y();
 }
 
-void ReadMapVarXML(luaTable& lt, ironfist_map::table_t table) {
-
-	luaTableElements lte;
-	luaSubTables lst;
+luaTable *ReadMapVarXML(ironfist_map::table_t table) {
 
+	luaTable *lt = new luaTable;
 	for (ironfist_map::table_t::tableElement_const_iterator tableE = table.tableElement().begin();
 		tableE != table.tableElement().end(); tableE++) {
-		lte[tableE->key().get()] = std::pair<MapVarType, std::string>(StringToMapVarType(tableE->type()), tableE->value().get());
-	}
+		mapVariable *mapVar = new mapVariable;
+		mapVar->type = StringToMapVarType(tableE->type());
 
-	for (ironfist_map::table_t::subTable_const_iterator subTable = table.subTable().begin();
-		subTable != table.subTable().end(); subTable++) {
-		lst.push_back(subTable->tableId().get());
-		ReadMapVarXML(lt, *subTable);
+		std::string *sV = new std::string(tableE->value().get());
+		mapVar->singleValue = sV;
+		(*lt)[tableE->key().get()] = *mapVar;

```

Comment:
This looks like a memory leak. You're allocating new memory for mapVar, copying it into the table, and....the pointer stays around.